app/Models/Atc/Endorsement.php

```diff
@@ -38,4 +38,9 @@ public function conditionsMetForUser(Account $account): bool
 
         return $allMet;
     }
+
+    public static function generateCacheKey($endorsementId, $accountId)
```

Comment:
Strong types on the parameters?

Reply:
I'd agree and perhaps convert to `$endorsement` and `$account` to pass in the full object

Reply:
> I'd agree and perhaps convert to `$endorsement` and `$account` to pass in the full object

If we did this, in the `FlushEndorsementCache` listener, we would have to do `Endorsement::get(['id'])->...` so that an instance of the Endorsement is created. Happy if I just leave it as passing in a full account model, but just the endorsement id?

Reply:
Sure that works

Reply:
Think I have found a solution that is a better combination of both!